src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Validation.cs

```diff
@@ -122,18 +124,25 @@ internal static unsafe partial class Utf8Utility
 
                         do
                         {
-                            if (Sse2.IsSupported)
+                            // pInputBuffer is 32-bit aligned but not necessary 128-bit aligned, so we're
+                            // going to perform an unaligned load. We don't necessarily care about aligning
+                            // this because we pessimistically assume we'll encounter non-ASCII data at some
+                            // point in the not-too-distant future (otherwise we would've stayed entirely
+                            // within the all-ASCII vectorized code at the entry to this method).
+                            if (AdvSimd.Arm64.IsSupported && BitConverter.IsLittleEndian)
+                            {
+                                mask = (uint)AdvSimd.Arm64.MaxAcross(AdvSimd.LoadVector128((byte*)pInputBuffer)).ToScalar();
```

Comment:
It seems this line may be causing the test issue. I need to figure out what is the correct way of creating the mask. It needs to be used afterwards (in the goto label).

Reply:
What `MoveMask` does on x86 is it takes the most significant bit from each element and packs them into the created `int32` result (that is the highest bit of element 0 becomes bit 0, the highest bit of element 1 becomes bit 1, etc).

So, if you need the exact match, you can mask out all but the most significant bit (using `And`) and then shift each element by a different amount so the bit is in the right position. e.g. for `Vector128<uint>` you would shift right element 0 by 31, element 1 by 30, element 2 by 29, and element 3 by 28
You would then sum them together to get an `int32` mask matching what x86 produces.

In other cases, you might not need to shift and can just and the bits directly. This is viable when you know the vector elements are either "zero" or "all bits set" (such as if it is the result of a comparison).

Reply:
Eirik's custom MoveMask method seemed reasonable to use here too. Can you please confirm? I used it in my latest commit.

Reply:
Yes, that looks like it will also produce a correct result